src/ledger/LedgerManagerImpl.cpp

```diff
@@ -427,55 +427,39 @@ LedgerManagerImpl::valueExternalized(LedgerCloseData const& ledgerData)
         assert(false);
     }
 
-    switch (closeLedgerIf(ledgerData))
-    {
-    case CloseLedgerIfResult::CLOSED:
-    {
-        setState(LM_SYNCED_STATE);
-        mApp.getCatchupManager().processLedger(ledgerData);
-    }
-    break;
-    case CloseLedgerIfResult::TOO_OLD:
-        // nothing to do
-        break;
-    case CloseLedgerIfResult::TOO_NEW:
-    {
-        setState(LM_CATCHING_UP_STATE);
-        mApp.getCatchupManager().processLedger(ledgerData);
-    }
-    break;
-    }
+    closeLedgerIf(ledgerData);
+
+    mApp.getCatchupManager().processLedger(ledgerData);
 }
 
-LedgerManagerImpl::CloseLedgerIfResult
+void
 LedgerManagerImpl::closeLedgerIf(LedgerCloseData const& ledgerData)
 {
     if (mLastClosedLedger.header.ledgerSeq + 1 == ledgerData.getLedgerSeq())
     {
-        if (mLastClosedLedger.hash ==
-            ledgerData.getTxSet()->previousLedgerHash())
+        auto& cm = mApp.getCatchupManager();
+        // if catchup work is running, we don't want ledger manager to close
+        // this ledger and potentially cause issues.
+        if (cm.isCatchupInitialized() && !cm.catchupWorkIsDone())
         {
-            closeLedger(ledgerData);
             CLOG(INFO, "Ledger")
-                << "Closed ledger: " << ledgerAbbrev(mLastClosedLedger);
-            return CloseLedgerIfResult::CLOSED;
-        }
-        else
-        {
-            CLOG(FATAL, "Ledger") << "Network consensus for ledger "
-                                  << mLastClosedLedger.header.ledgerSeq
-                                  << " changed; this should never happen";
-            CLOG(FATAL, "Ledger") << POSSIBLY_CORRUPTED_QUORUM_SET;
-            throw std::runtime_error("Network consensus inconsistency");
+                << "Can't close ledger: " << ledgerAbbrev(mLastClosedLedger)
+                << "  in LM because catchup is running";
+            return;
         }
+
+        closeLedger(ledgerData);
+        CLOG(INFO, "Ledger")
+            << "Closed ledger: " << ledgerAbbrev(mLastClosedLedger);
+
+        setState(LM_SYNCED_STATE);
```

Comment:
I think it makes sense to mark ledger manager as "synced" when the last closed ledger is the newest ledger we know of. This is not the case when LM has seen some future ledger (at which point CatchupManager has started buffering ledgers). So I think when catchup has buffered some newer ledgers, LM should stay in "catching up" state.

Reply:
Done. I ended up keeping the requirement to close the next ledger before moving to the synced state so the logic stays consistent with how catchup works.